### PR TITLE
Add SDK location unit tests

### LINKED ISSUE #46137: Add xcode sdkLocation unit tests
Missing unit tests for `sdkLocation()` in xcode.dart.
Suggested in PR https://github.com/flutter/flutter/pull/46130#discussion_r354064169

### PR DESCRIPTION
**Pull request by jmagman**
## Description

Add unit tests.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/46137

## Tests

xcode_tests

## Checklist
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

### CODE REVIEW

**Comment by jmagman** [comment]
Just the unit tests in https://github.com/flutter/flutter/pull/52570 without the SDK env lookup behavior changes.

File: packages/flutter_tools/test/general.shard/macos/xcode_test.dart [hunk 1]
```diff
@@ -359,18 +385,6 @@ void main() {
         Platform: () => macPlatform,
       });
 
-      testWithoutContext('uses timeout', () async {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/macos/xcode_test.dart]: Why was this test removed?

Comment (jmagman) [reply packages/flutter_tools/test/general.shard/macos/xcode_test.dart]: Uhhhhh whoops, rebase issue.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**